### PR TITLE
Tighten the CountedBitSet class

### PR DESCRIPTION
**Pull request by dnhatn**
Sorry @jasontedor for missing your comments in https://github.com/elastic/elasticsearch/pull/27547. This PR addresses those comments.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/translog/CountedBitSetTests.java [hunk 1]
```diff
@@ -79,6 +82,7 @@ public void testReleaseInternalBitSet() {
             assertThat(countedBitSet.isInternalBitsetReleased(), equalTo(true));
             assertThat(countedBitSet.length(), equalTo(numBits));
             assertThat(countedBitSet.cardinality(), equalTo(numBits));
+            assertThat(countedBitSet.ramBytesUsed(), lessThan(ramBytesUsedWithBitSet));
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/index/translog/CountedBitSetTests.java]: Make the constant package private and assert that the size here is `BASE_RAM_BYTES_USED`?

**Review by jasontedor — APPROVED** [review]
I left one suggestion but either way LGTM.

**Comment by jasontedor** [comment]
Thanks @dnhatn!

File: core/src/test/java/org/elasticsearch/index/translog/CountedBitSetTests.java (hunk 1 above)
Comment (dnhatn) [reply core/src/test/java/org/elasticsearch/index/translog/CountedBitSetTests.java]: I pushed https://github.com/elastic/elasticsearch/pull/27632/commits/8dbf71dedccbaa0878cb46b3b0782d9af846abb9

**Comment by dnhatn** [comment]
Thanks @jasontedor.

**Pull request merged**